tutor/commands/config.py

```diff
@@ -220,8 +224,32 @@ def patches_list(context: Context) -> None:
     renderer.print_patches_locations()
 
 
+@click.command(name="edit", help="Edit config.yml of the current environment")
+@click.pass_obj
+def edit(context: Context) -> None:
+    config_file = os.path.join(context.root, "config.yml")
+
+    open_cmd = None
+
+    if which("open"):  # MacOS
```

Comment:
Let's amend the comment. `open` is also available on Ubuntu, for instance.

I have another, more important problem with this command. According to my testing, when I run `open file.txt` on a headless server (for instance: a remote production server), it opens a read-only editor. How can we fix this command to open a write-capable editor? (`xdg-open` is not available on servers without a Desktop) Note that `$EDITOR` is set to "vim" on this server.

Reply:
@regisb I wanted to test this scenario to evaluate the options. However, the server image I used (ubuntu based) didn't have a `open` at all. And my desktop with Ubuntu seems to just ship `xdg-open` as `open`. 

```
❯ which open
/usr/bin/open

❯ open --version
xdg-open 1.1.3
```

> when I run `open file.txt` on a headless server (for instance: a remote production server), it opens a read-only editor.

Which version of `open` is this, and what's the editor that it opens?

Reply:
On the topic of a "writable editor", MacOS's `open` implements `-a` flag, which allows specifying the application to open the file in. I am wondering if something similar can be added to this command which can force a specific editor to be launched. `tutor config edit -a vim` maybe? What do think about this?

Reply:
I think this can be a good option, this gives control to the user. For instance, if someone wants to open the file in Sublime/VScode, they can override the default utils the command is checking.